### PR TITLE
Fronts on Articles 

### PR DESCRIPTION
**Pull request by holder-josh**
The initial AB test went well, so this is a V2 of that. Much more of a 'production ready' test that obeys the outbrain rules and injects a mixture of section and network fronts.

Set up as a 0% test for now.

To do in another pull request:
- Sort out the 'Jump to comments' button's behaviour (currently jumps a lot due to AJAXing components in)
- Sort tracking

### CODE REVIEW

**Comment by Jholder112233** [comment]
@johnduffell one for you to take a look at :dancers:

File: common/app/views/support/ContentFooterContainersLayout.scala [hunk 1]
```diff
@@ -28,7 +30,7 @@ object ContentFooterContainersLayout {
 
       def includeOutbrainPlaceholder(htmlBlocks: Seq[Html]): Seq[Html] = {
         if (content.showFooterContainers && !content.shouldHideAdverts && OutbrainSwitch.isSwitchedOn) {
-          val pos = if ((content.isSeries || content.isBlog) && !related.hasStoryPackage) {
+          val pos = if (((content.isSeries || content.isBlog) && !related.hasStoryPackage) || (!content.showInRelated && !related.hasStoryPackage)) {
```
Comment (johnduffell) [inline common/app/views/support/ContentFooterContainersLayout.scala]: might be worth making a couple of vals to make this make more sense
```
val seriesWithRelated = (content.isSeries || content.isBlog) && !related.hasStoryPackage
val hasRelatedButNotShown = !content.showInRelated && !related.hasStoryPackage
```

File: facia/app/controllers/FaciaController.scala [hunk 2]
```diff
@@ -45,6 +46,40 @@ trait FaciaController extends Controller with Logging with ExecutionContexts wit
   def renderFrontJson(id: String) = renderFront(id)
   def renderContainerJson(id: String) = renderContainer(id, false)
 
+  def renderSomeFrontContainers(path: String, num: String, offset: String, size: String) = MemcachedAction { implicit request =>
+    getSomeCollections(Editionalise(path, Edition(request)), num.toInt, offset.toInt).map { collections =>
```
Comment (johnduffell) [inline facia/app/controllers/FaciaController.scala]: what is num (or offset) isn't an int?  should we use a match statement and return a bad request 400 if not?

File: facia/app/controllers/FaciaController.scala [hunk 3]
```diff
@@ -262,6 +297,12 @@ trait FaciaController extends Controller with Logging with ExecutionContexts wit
       })
     }.getOrElse(successful(None))
 
+  private def getSomeCollections(path: String, num: Int, offset: Int = 0): Future[Option[List[PressedCollection]]] =
+      frontJsonFapi.get(path).map(_.flatMap{ faciaPage =>
+        // To-do: change the filter to only exclude thrashers and empty collections, not items such as the big picture
+        Some(faciaPage.collections.filterNot(collection => (collection.curated ++ collection.backfill).length < 2 || collection.displayName == "most popular").drop(offset).take(num))
```
Comment (johnduffell) [inline facia/app/controllers/FaciaController.scala]: is display name the only way to tell?

Comment (johnduffell) [inline facia/app/controllers/FaciaController.scala]: might be possible to improve the formatting here a bit, splitting each conceptual step onto a separate line

File: static/src/javascripts/projects/common/modules/onward/fronts-containers.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    function FrontsContainers() {
+        moveComments();
+        insertFirstTwo();
+        insertFinalTwo();
+    }
+
+    function insertFirstTwo() {
+        var front = (loadSection) ? config.page.section : edition;
+
+        insertContainers(front, $('.js-fronts-containers'), 2, 0, 'original', function () {});
```
Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/onward/fronts-containers.js]: do we only ever use 'original' on this endpoint?

**Comment by johnduffell** [comment]
:+1:

**Comment by Jholder112233** [comment]
@kelvin-chappell do you mind taking a look at the changes to ContentFooterContainersLayout and ContentFooterContainersLayoutTest? Seems like you were the last person to touch those. Would appreciate a sanity check on the changes

**Comment by kelvin-chappell** [comment]
It looks fine.  The main thing is that the outbrain component has to appear as the second container below the article I think. :+1:

**Comment by Jholder112233** [comment]
@kelvin-chappell thank you, from my testing it still does. Will ship and test it thoroughly again.

**Pull request merged**